.github/workflows/build_steps.yml

```diff
@@ -73,6 +73,13 @@ jobs:
         with:
           submodules: recursive # Just in case a dep has its own third-party deps
 
+      # See: https://github.com/actions/runner-images/issues/9680#issuecomment-2051917949
+      - name: HOTFIX Setup CMake 3.29.2
```

Comment:
We don't really need to say `HOTFIX` do we? Choosing our own CMake seems sensible and not a hack.

Why are we only doing this on Windows? Using the CMake across all platforms might help to simplify the CI setup a bit.

Reply:
The next image should come with this newer version of CMake, and I think this pin should rather be temporary as it duplicates installations of CMake.

I pinned for Windows because it was the only OS affected by the update of the image.

Nevertheless, the name of this step definitely can be changed.

Feel free to author changes on top of mine if you want to.

Reply:
What do you think?
```suggestion
      - name: Setup CMake 3.29.2
```

Reply:
Merging as is.